topology-machine/topomachine

```diff
@@ -187,41 +215,98 @@ def run_topology(config, dry_run, with_trace=False):
-            output,_ = run_command(["docker", "inspect", "--format", "{{.State.Health.Status}}", name])
-            print("Container already running. Health: %s" % output.strip())
+        state_output = ''
+        try:
+            state_output = get_container_state_api(url, name)
+        except:
+            pass
+        if state_output:
+            health_output = ''
+            try:
+                health_output = get_container_health_api(url, name)
+            except:
+                pass
+            print("Container(%s) already running. Health: %s" % (name, health_output.strip()))
         else:
-            run_command(cmd, dry_run)
+            print("Starting container(%s)" % name)
+            run_container_api(url, image, name, api_cmd)
 
     if 'links' in config:
         name = "%svr-xcon" % args.prefix
-        cmd = ["docker", "run", "--privileged", "-d", "--name", name]
+        image = "%s%s" % (docker_registry, 'vr-xcon')
 
+        links = []
+        cmd = []
         for vr in sorted(config['routers']):
-            cmd.extend(["--link", "%s%s:%s" % (args.prefix, vr, vr)])
-        cmd.append(docker_registry + "vr-xcon")
+            links.append((args.prefix + vr, args.prefix + vr))
         cmd.append("--p2p")
         cmd.extend(["%s%s/%s--%s%s/%s" % (args.prefix, link["left"]["router"],
                                            link["left"]["numeric"],
                                            args.prefix,
                                            link["right"]["router"],
                                            link["right"]["numeric"]) for link in config['links']])
-        output,_ = run_command(["docker", "inspect", "--format", "{{.State.Running}}", name])
-        if output.strip() == "true":
-            output,_ = run_command(["docker", "inspect", "--format", "{{.State.Health.Status}}", name])
-            print("Container already running. Health: %s" % output.strip())
+        state_output = ''
+        try:
+            state_output = get_container_state_api(url, name)
+        except:
+            pass
+        if state_output:
+            health_output = ''
+            try:
+                health_output = get_container_health_api(url, name)
+            except:
+                pass
+            print("Container(%s) already running. Health: %s" % (name, health_output.strip()))
         else:
-            run_command(cmd, dry_run)
+            print("Starting container(%s)" % name)
+            run_container_api(url, image, name, cmd, links)
+
+
+
+def stop_topology_api(config, url):
+
+    if 'routers' not in config:
+        print("No routers in config")
+        sys.exit(1)
+
+    for router in sorted(config['routers']):
+        name = "%s%s" % (args.prefix, router)
+        try:
+            stop_container_api(url, name)
+            print("Stopped container(%s)" % name)
+        except:
+            print("Container(%s) is not running" % name)
+            pass
```

Comment:
Superfluous `pass` since there's a print statement.

I'd also prefer if we would catch a more specific exception and not everything... :)